doc/cabal-package.rst

```diff
@@ -931,34 +939,26 @@ look something like this:
 
     library
         exposed-modules:  Foo.Public
-        build-depends:    foo-internal, base >= 4.3 && < 5
+        build-depends:    foo:foo-internal, base >= 4.3 && < 5
         default-language: Haskell2010
 
     test-suite test-foo
         type:             exitcode-stdio-1.0
         main-is:          test-foo.hs
         -- NOTE: no constraints on 'foo-internal' as same-package
         --       dependencies implicitly refer to the same package instance
-        build-depends:    foo-internal, base
+        build-depends:    foo:foo-internal, base
         default-language: Haskell2010
 
-Sublibraries are also useful for packages that define multiple
-executables, but do not define a publicly accessible library. Internal
-libraries are only visible internally in the package (so they can only
-be added to the :pkg-field:`build-depends` of same-package libraries,
-executables, test suites, etc.) Sublibraries locally shadow any
-packages which have the same name; consequently, don't name an internal
-library with the same name as an external dependency if you need to be
-able to refer to the external dependency in a
-:pkg-field:`build-depends` declaration.
+Another example of the use of internal libraries is a package that includes one
```

Comment:
I feel like this cuts quite a bit. Some of this overlaps with the 'visibility' field, but it's good to have this info here as well I think. Especially the explanation of shadowing seems important.

Reply:
On shadowing, I thought it no longer applies (from Cabal 3.4), because a sublibrary is now specified in a `build-depends:` as `package_name:library_name`. I think shaddowing was only applicable when a sublibrary could be specified as just `library_name` (which might clash with a package's name).

On the rest that is cut, it is all directly duplicative.

Reply:
On Cabal 3.4, see: https://cabal.readthedocs.io/en/stable/file-format-changelog.html#cabal-version-3-4:
> Dependencies to sublibraries must be specified explicitly, even for current package. For example: build-depends: mypackage:internal-lib This way you can have an internal library with the same name as some external dependency.

Reply:
'Shadowing' brought back in, elsewhere.